redbpf/src/ringbuf.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 Authors of Red Sift
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! # Ring Buffer Event handling
+//!
+//! The ringbuf module makes it easier to hook up and consume from ring buffers.
+//! and provdes a safe interface for accesing the ring buffer
+//!
+//! The consumed data is a raw pointer and will require unsafe code to transform
+//! into a data structure.
+
+use crate::{Error, Map, Result};
+use std::cell::RefCell;
+use std::io;
+use std::os::unix::io::RawFd;
+use std::slice;
+
+use bpf_sys::{
+    ring_buffer, ring_buffer__consume, ring_buffer__epoll_fd, ring_buffer__free, ring_buffer__new,
+};
+use libc::c_void;
+
+#[derive(Debug)]
+pub struct RingBufMapContext {
```

Comment:
Do you expect this API will be used by users directly?

Reply:
This should not be public. Thanks!

Reply:
Fixed.